core/unit_test/TestGraph.hpp

```diff
@@ -101,6 +101,15 @@ struct TEST_CATEGORY_FIXTURE(graph) : public ::testing::Test {
     Kokkos::deep_copy(ex, bugs, 0);
     ex.fence();
   }
+
+  template <typename Exec>
+  void check(const Exec& exec, const int expt_count, const int expt_bugs) {
+    Kokkos::deep_copy(exec, count_host, count);
+    Kokkos::deep_copy(exec, bugs_host, bugs);
+    exec.fence();
+    ASSERT_EQ(expt_count, count_host());
+    ASSERT_EQ(expt_bugs, bugs_host());
+  }
```

Comment:
OK with abstracting away but this approach has a big downside that is that failure reports will always point to that function instead of the call site.
Maybe a macro could help.
Also you can probably do better for the name.

And you probably don't want to tie the fate of this PR to that (somewhat unrelated) change, do you?

Reply:
Clearly, if we want the call site to be clearly identified, we'll need to go through a macro. Otherwise, it will always point to the `check` method location (which is not bringing much information).

Reply:
@dalg24  OK to close ?